app/models/data_file.rb

```diff
@@ -26,6 +24,14 @@ class DataFile < Container
 
   delegate :http_verb, to: :upload
 
+  def upload=(val)
+    @upload=val
```

Comment:
I have a great idea! Why dont we get rid of the callbacks for setting the current_version, and just use the upload setter to set the current file version :P